File: ClaimPortalJava/src/main/java/com/mt/claimportaljava/models/Policy.java
package com.mt.claimportaljava.models;


import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name = "policy")
public class Policy {
    @Id
    @GeneratedValue
    private Long id;

    private String policyNumber;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "policyholder_id", referencedColumnName = "id")
    private Contact policyHolder;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "insured_id", referencedColumnName = "id")
    private Contact insured;

    private Date effectiveDate;

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public String getPolicyNumber() {
        return policyNumber;
    }

    public void setPolicyNumber(String policyNumber) {
        this.policyNumber = policyNumber;
    }

    public Contact getPolicyHolder() {
        return policyHolder;
    }

    public void setPolicyHolder(Contact policyHolder) {
        this.policyHolder = policyHolder;
    }

    public Contact getInsured() {
        return insured;
    }

    public void setInsured(Contact insured) {
        this.insured = insured;
    }

    public Date getEffectiveDate() {
        return effectiveDate;
    }

    public void setEffectiveDate(Date effectiveDate) {
        this.effectiveDate = effectiveDate;
    }
}

File: ClaimSystemPython/src/models/policy.py
from sqlalchemy import Column, Integer, String, ForeignKey, Date
from sqlalchemy.orm import MappedColumn, Relationship

from src.models.base import Base


class Policy(Base):
    __tablename__ = 'Policy'

    id = Column(Integer, primary_key=True)
    policy_number = Column(String(20))
    policyholder_id = MappedColumn(ForeignKey('Contact.id'))
    policyholder = Relationship('Contact', cascade="all", foreign_keys=[policyholder_id])
    insured_id = MappedColumn(ForeignKey('Contact.id'))
    insured = Relationship('Contact', cascade="all", foreign_keys=[insured_id])
    effective_date = Column(Date)
